File: domain/src/main/java/com/tonitealive/app/domain/model/ApiError.java
package com.tonitealive.app.domain.model;

import com.google.auto.value.AutoValue;
import com.google.gson.annotations.SerializedName;

@AutoValue
public abstract class ApiError {

    @SerializedName("timestamp") abstract Long timestamp();
    @SerializedName("status") abstract int status();
    @SerializedName("error") abstract String error();
    @SerializedName("exception") abstract String exception();
    @SerializedName("message") abstract String message();
    @SerializedName("path") abstract String path();
    @SerializedName("error_code") abstract int errorCode();


}

File: domain/src/main/kotlin/com/tonitealive/app/domain/model/ApiError.kt
package com.tonitealive.app.domain.model

import com.google.gson.annotations.SerializedName


data class ApiError (@SerializedName("timestamp") val timestamp: Long,
                     @SerializedName("status") val status: Int,
                     @SerializedName("error") val error: String,
                     @SerializedName("exception") val exception: String,
                     @SerializedName("message") val message: String,
                     @SerializedName("path") val path: String)